app/models/topic.rb

```diff
@@ -1039,6 +1044,43 @@ def invite_existing_muted?(target_user, invited_by)
     false
   end
 
+  def allowed_pm_user?(target_user, invited_by)
+    return true if target_user.staff?
+    users = TopicAllowedUser.where(topic: self).pluck(:user_id)
```

Comment:
Could we also return early if `invited_by.staff?`

Reply:
Actually I think not? Here is a scenario: A conversation between StaffA and UserB has started. UserB has allowed_pm_users enabled and only has allowed StaffA and UserD. StaffA invites UserC, but UserB hasn't allowed them so they can't be added to the pm.

Reply:
That is an interesting edge case. I kinda feel like staff should be able to say "do this anyway" but I also see your point of view. Let's keep it your way and see how it goes.